agora/Agora/Proposal.hs

```diff
@@ -717,13 +722,13 @@ pisEffectsVotesCompatible ::
   forall (s :: S).
   Term
     s
-    ( PMap 'Unsorted PResultTag PProposalEffectGroup
+    ( PMap 'Sorted PResultTag PProposalEffectGroup
         :--> PProposalVotes
         :--> PBool
     )
 pisEffectsVotesCompatible = phoistAcyclic $
   plam $ \m (pto -> v :: Term _ (PMap _ _ _)) ->
-    PUM.pkeysEqual # m # v
+    PSM.pkeysEqual # m #$ punsafeCoerce v
```

Comment:
Shouldn't we require `PProposalVotes` to be sorted also? Since they are supposed to be homomorphic? It would get rid of this `punsafeCoerce` here.

Reply:
`PSM.pkeysEqual` performs a list equal check internally, and we have already ensure that `m` is sorted, so imo it's safe to just `punsafeCoerce` here. And also note that we can't really change the inner type of `PProposalVote` cause otherwise we won't be able to lift it.

Reply:
Right, inability to lift is very annoying, you're right. This is suitable.